lib/git_service/commit.rb

```diff
@@ -7,7 +7,7 @@ def initialize(rugged_repo, commit_oid)
     end
 
     def diff(other_ref = parent_oids.first)
-      Diff.new(rugged_diff(other_ref))
+      @diff ||= Diff.new(rugged_diff(other_ref))
```

Comment:
This should not be memoized because the diff is dependent on the paramter

Reply:
Or if it is memoized, then we need some sort of Hash by `other_ref`

Reply:
Yeah, that is fair.  Maybe I will just locally memoize then (or not bother and drop that commit), but basically `full_message` and `details_hash` call this method twice, so was trying to be a good "performance developer".

We will see, but I might just lean towards dropping it based no your comment.

Reply:
Dropped.  Resolving.